testsuite/single_node_performance.py

```diff
@@ -74,16 +76,132 @@
 
 
 def execute_command(command):
-    try:
-        output = subprocess.check_output(
-            command, shell=True, text=True, cwd=target_directory
+    result = []
+    with Popen(
+        command,
+        shell=True,
+        text=True,
+        cwd=target_directory,
+        stdout=PIPE,
+        bufsize=1,
+        universal_newlines=True,
+    ) as p:
+        # stream to output while command is executing
+        for line in p.stdout:
+            print(line, end="")
+            result.append(line)
```

Comment:
why not just `capture_output=True`?

Reply:
command runs for a while, and with capture_output=True we are not able to see incremental progress. 

Also if there is a deadlock, nothing will be printed.